CRM/Utils/Check/Component/DedupeRules.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
+class CRM_Utils_Check_Component_DedupeRules extends CRM_Utils_Check_Component {
+
+  /**
+   * Get dedupe rules, grouped by contact type, for a specific usage
+   *
+   * @param string $used (Supervised, Unsupervised, General)
+   * @return string[]
+   */
+  private static function getContactTypesForRule($used) {
+    $dedupeRules = \Civi\Api4\DedupeRuleGroup::get()
+      ->addSelect('contact_type')
+      ->setGroupBy([
+        'contact_type',
+      ])
+      ->addWhere('used', '=', $used)
+      ->execute();
+
+    $types = [];
+    foreach ($dedupeRules as $rule) {
+      if (!in_array($rule, $types)) {
+        $types[] = $rule['contact_type'];
+      }
+    }
+    return $types;
+  }
+
+  /**
+   * Returns an array of missing expected contact types
+   *
+   * @param string[] $rules
+   * @return array
+   */
+  private static function getMissingRules($rules) {
+    $types = ['Organization', 'Household', 'Individual'];
+    return array_diff($types, $rules);
```

Comment:
I just realized this function is returning a value which needs to be output to the user, so translatable labels need to be returned, not just machine names. This should fix it.

```suggestion
    $types = array_column(CRM_Contact_BAO_ContactType::basicTypeInfo(), 'label', 'name');
    return array_diff_key($types, array_flip($rules));
```